pulsar/consumer_partition.go

```diff
@@ -597,7 +602,18 @@ func (pc *partitionConsumer) MessageReceived(response *pb.CommandMessage, header
 			pc.AckID(msgID)
 			continue
 		}
-
+		var messageIndex *uint64
+		var brokerPublishTime *time.Time
```

Comment:
Maybe instead of using pointers we can use the following definition:

```
		var messageIndex uint64
		var brokerPublishTime time.Time
```

Reply:
@wolfstudy It can be null if broker not enable `brokerMetadata` feature